sdk/digitaltwins/azure-digitaltwins-core/API design.md

```diff
@@ -525,6 +524,358 @@ These APIs are invoked via DigitalTwinsClient.
 ```
 </details>
 
+## Digital Twins
+<details><summary><b>Terminology</b></summary>
+
+A digital twin is an instance of one of your custom-defined models.
+
+</details>
+
+<details><summary><b>Async APIs</b></summary>
+
+These APIs are invoked via DigitalTwinsAsyncClient.
+
+```java
+/**
+ * Gets a digital twin.
+ *
+ * @param digitalTwinId The Id of the digital twin. The Id is unique within the service and case sensitive.
+ * @return The application/json digital twin
+ */
+@ServiceMethod(returns = ReturnType.SINGLE)
+public Mono<string> getDigitalTwin(String digitalTwinId)
+
+/**
+ * Gets a digital twin.
+ *
+ * @param digitalTwinId The Id of the digital twin. The Id is unique within the service and case sensitive.
+ * @param modelClass The model class to convert the response to.
+ * @return A Http response containing the application/json digital twin
+ */
+@ServiceMethod(returns = ReturnType.SINGLE)
+public <T> Mono<T> getDigitalTwinWithResponse(String digitalTwinId, Class<T> modelClass)
```

Comment:
modelClass is a good name, but it's may be confused with actual ADT models, 
we can use something like klazz/clazz (standard) or `classType `or something

Reply:
That was my concern as well, although I believe I recommended that we use `modelClass`. 
Since every type that we want to convert to is essentially a "model" in programming terms, I thought this should be ok, although I can see the potential for confusion with ADT models.

I think `classType` is well descriptive, we can use that. (I am sure the SDK board will push back if that goes against java naming conventions. )

Reply:
we will cross that bridge when it comes to reviewing it with the board I guess.

Reply:
I'll also take a look at other sdks otherwise classType looks fine to me